### PR TITLE
feature/cplp-1378 app document update

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1378

### CODE REVIEW

**Comment by jjeroch** [comment]
@qxz2mqe  can you please fix the security hotspot. Its not a critical finding, but would be good to still fix it or to include and comment in case this is a false "positive"

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status404NotFound)]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status415UnsupportedMediaType)]
+    public async Task<IActionResult> UpdateAppDocumentAsync([FromRoute] Guid appId, [FromRoute] DocumentTypeId documentTypeId, [FromForm(Name = "document")] IFormFile document)
+    {
+        await this.WithIamUserId(userId => _appReleaseBusinessLogic.UpdateAppDocumentAsync(appId,  documentTypeId,  document, userId)).ConfigureAwait(false);
+        return NoContent();
+    }
+}
```
Comment (Phil91) [inline src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs:100]: please add a line break

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using CatenaX.NetworkServices.PortalBackend.PortalEntities.Enums;
+using CatenaX.NetworkServices.PortalBackend.PortalEntities.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace CatenaX.NetworkServices.PortalBackend.DBAccess.Repositories;
+
+/// <summary>
+/// Implementation of <see cref="IAppReleaseRepository"/> accessing database with EF Core.
+/// </summary>
+public class AppReleaseRepository : IAppReleaseRepository
```
Comment (Phil91) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs:32]: I'd suggest moving the GetAppByIdAsync to the AppRepository, maybe rename it with ...AsAppUpdateModel.
and rename the AppReleaseRepository to AppAssignedDocumentsRepository.

**Review by Phil91 — CHANGES_REQUESTED** [review]
As discussed in the review, let's wait with creating the Migration for the database change until the pending database changes are merged

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs:100]: Could you please provide me the line number after which i have to give a line break

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/portalbackend/CatenaX.NetworkServices.PortalBackend.DBAccess/Repositories/AppReleaseRepository.cs:32]: Moved GetAppByIdAsync to the AppRepository, All the method/action i am writing in AppReleaseRepository  is related to appreleaseprocess so i think naming is compatible with the feature we want to implement

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs (hunk 1 above)
Comment (Phil91) [reply src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs]: sorry, was at the end of the file

**Review by Phil91 — APPROVED** [review]

**Review by Phil91 — CHANGES_REQUESTED** [review]
Just recognized that there are some code smells, could you take a look at them

**Comment by qxz2mqe** [comment]
Yes I am looking into that but could you please let me know the timing to review script for 1378 because you have already merged the code into main. If some one pull the latest and don’t find those attribute then they might get ecveption

**Comment by Phil91** [comment]
> 
I will. We need the migration as well

**Review by Phil91 — COMMENTED** [review]
Everything looks fine so far. The migration is missing, after that we can update devenv and merge this pr

**Comment by jjeroch** [comment]
added @Akshay-Chaudhari-BMW  please inform him as soon as pr is merged, since this merge and deployment is needed for akshay to proceed on his ticcket

**Comment by qxz2mqe** [comment]
https://github.com/tractusx-team-portal-onboarding/devenv/pull/111   PR Created to review script

**Review by Phil91 — APPROVED** [review]
Looks fine :)

**Pull request merged**